File: 프로그래머스/lv2/76502. 괄호 회전하기/괄호 회전하기.js
function solution(s) {
    var answer = 0;
    for (let i = 0; i < s.length; i++) {
        let arr = s.slice(i) + s.slice(0,i);
        let stack = [];
        for (let j of arr) {
            stack.push(j);
            let par = stack[stack.length - 2] + stack[stack.length - 1];
            if (par === '{}' || par === '[]' || par ==='()') {
                stack.pop();
                stack.pop();
            }
        }
        if (stack.length === 0) answer++;
    }
    return answer;
}

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
def solution(s):
    answer = 0
    for i in range(len(s)):
        arr = s[i:] + s[:i]
        stack = []
        for j in arr:
            stack.append(j)
            if len(stack) >= 2:
                par = stack[-2] + stack[-1]
                if par in ['{}', '[]', '()']:
                    stack.pop()
                    stack.pop()
        if len(stack) == 0:
            answer += 1
    return answer